Calculator_exercise.py

```diff
@@ -0,0 +1,299 @@
+"""Calculator exercise"""
+
```

Comment:
A general comment on your approach. You are calling a function ("space remover", which calls a "parenthesis" which calls factorial). The problem with this is that it makes finding something specific in your code very difficult for anyone reading it (and will also be way harder to debug).

Your code will be much clearer if you have "handoffs" within some main evaluate
To be clear. You want something like

def evaluate(expression):
         expression = remove_spaces(expression)
         expression = handle_parenthesis(expression)
         expression = handle_factorials(expression)

this allows me to easily see the various steps in your code and then I can look at the function implementation if I want more details.

You can also do a more object oriented approach. Something like

def calculate(input):
            my_expression = Expression(input)
            my_expression.remove_spaces()
            my_expression.remove_parentheses()
            my_expression.handle_factorials()

For this exercise (I think) both are considered valid approaches. In general we prefer the object oriented approach as it is cleaner and easier to follow when problems get bigger.